contracts/utils/cryptography/ECDSA.sol

```diff
@@ -180,10 +180,14 @@ library ECDSA {
      *
      * See {recover}.
      */
-    function toEthSignedMessageHash(bytes32 hash) internal pure returns (bytes32) {
-        // 32 is the length in bytes of hash,
+    function toEthSignedMessageHash(bytes32 hash) internal pure returns (bytes32 message) {
         // enforced by the type signature above
-        return keccak256(abi.encodePacked("\x19Ethereum Signed Message:\n32", hash));
+        /// @solidity memory-safe-assembly
+        assembly {
+            mstore(0, 0x19457468657265756d205369676e6564204d6573736167653a0a333200000000)
```

Comment:
We should be able to write as follows:

```suggestion
        bytes32 prefix = "\x19Ethereum Signed Message:\n32";
        /// @solidity memory-safe-assembly
        assembly {
            mstore(0, prefix)
```

Reply:
What about this?
```
    function toEthSignedMessageHash(bytes32 hash) external view returns (bytes32 message, uint256 gasUsed) {
        assembly {
            let gas0 := gas()
            mstore(0, "\x19Ethereum Signed Message:\n32")
            mstore(28, hash)
            message := keccak256(0, 60)
            gasUsed := sub(gas0, gas())
        }
    }
```
What you suggested costs gas a little more.
I think, this is better because it costs less gas and enough easy to understand!

Reply:
Yeah that works for me as well.